zerver/tests/test_decorators.py

```diff
@@ -1207,7 +1207,7 @@ def test_login_deactivated_user(self) -> None:
 
         result = self.login_with_return(self.example_email("hamlet"))
         self.assert_in_response(
-            "Your account is no longer active.",
+            "Your account {} is no longer active.".format(self.example_email("hamlet")),
```

Comment:
Since this is deactivation code, we probably want to hard-code the email into the expected response here.  Are we expecting `self.example_email("hamlet")` to be "hamlet@example.com" here or some mangled version of the email?  Particularly if it's mangled, we should make it explicit in the test.  Or if we don't want it mangled, we should extract the string first before going back to the DB to get it (so just do `hamlet_email = user_profile.email` before `do_deactivate_user`.)

Does that make any sense?

Reply:
We also want to make sure this is accessing `.delivery_email`.